### DISCUSSION TITLE
Bazel error on variable reference in the version attribute of maven.artifact block

### DISCUSSION THREAD

**Discussion by lyonlai** [renovatebot/renovate#30246] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

We are hitting a bazel `WORKSPACE` file dependency extracting issue. Where there is a block of `maven_install` file is having errors on extracting dependencies from a  `maven.artifact` block that refer its version to a shared variable another file like the following. This error is not thrown out but could be reproduced in the unit test.  This error causing no dependency extraction from this particular `maven_install` block

```bazel
        maven.artifact(
            artifact = "grpc-core",
            exclusions = [
                "io.grpc:grpc-util",
            ],
            group = "io.grpc",
            version = GRPC_VERSION,
        ),
```

at the top of the file the `GRPC_VERSION` is loaded from a separate `bzl` file. 

```
load("//grpc:versions.bzl", "GRPC_VERSION")
```

It seems like the bazel manager is not handling shared variable very well just yet. This variables is being used in other 10 artifacts that comes from `io.grpc` group. 

Other places that's used as string interpolation of `io.grpc:grpc-xyz:%s` also doesn't work properly as the version is `%s` and it's probably going to be ignored due to the invalid version scheme. But this is a separate issue. 

Wondering what's the best way forward on the variable reference in the version attribute of the `maven.artifact` block. 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Sounds like a moderately complex feature to add. Is `//grpc:versions.bzl` a local file?

**Comment by lyonlai** [comment]
It is a local file. this is what the inside looks like.

```bazel
GRPC_VERSION = "1.62.2"
```
